sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,286 @@
+
+#pragma once
+
+#include <seastar/parquet/parquet/column_writer.h>
+#include <mutation_fragment.hh>
+#include <tombstone.hh>
+#include <dht/i_partitioner.hh>
+#include <range_tombstone.hh>
+#include <seastar/parquet/parquet/file_writer.h>
+#include <seastar/parquet/parquet/schema.h>
+#include <seastar/parquet/parquet/properties.h>
+#include <seastar/core/sstring.hh>
+#include <schema.hh>
+#include <iostream>
+
+namespace parquet {
+
+// TODO will be removed
+std::ofstream _trace = std::ofstream("/tmp/parquet-trace.log", std::ofstream::binary);
+
+using pq_schema = std::shared_ptr<schema::GroupNode>;
+using row_buffer = std::unordered_map<column_id, std::vector<bytes>>;
```

Comment:
I store the raw linearized bytes in a vector, then I extract the value from them using deserialize method. An alternative can be storing ready to write values directly. A lot copying is involved here. I am sure there are ways to reduce it.

Reply:
It's fine to keep linearized bytes for now, because then we claim ownership of the data - if we were to store values, we'd need to make sure that their data still exist when we want to use them (since `value` is only a non-owning view over the data, just like `string_view` is to `string`). One thing that needs to be done is a couple of `std::move()` calls - e.g. when you create a `bytes` object and put it in the map, move the contents to make sure that we don't invoke copy constructors, but instead just move internal pointers:
```cpp
bytes b = something.deserialize();
_buffer[name].push_back(std::move(b));
```